lldb/source/Plugins/Platform/Android/PlatformAndroid.cpp

```diff
@@ -287,7 +287,7 @@ Status PlatformAndroid::DownloadModuleSlice(const FileSpec &src_file_spec,
   static constexpr llvm::StringLiteral k_zip_separator("!/");
   size_t pos = source_file.find(k_zip_separator);
   if (pos != std::string::npos)
-    source_file = source_file.substr(0, pos);
+    source_file = source_file.resize(0, pos);
```

Comment:
According to https://en.cppreference.com/w/cpp/string/basic_string/resize, `resize(pos)` would be correct here. Otherwise we're resizing to 0 characters.

Reply:
`std::string::resize()` returns `void` so this won't even compile. The right fix is:
```
source_file.resize(pos);
```

Reply:
This still isn't resolved, you're still doing `source_file = source_file.resize(pos)`. It doesn't return a value.

Reply:
Thanks, got it.